pkg/cache/cache.go

```diff
@@ -149,13 +149,13 @@ type ClusterQueue struct {
 	podsReadyTracking bool
 }
 
-func (cq *ClusterQueue) IsBorrowing() bool {
-	if cq.Cohort == nil || len(cq.Usage) == 0 {
+func (c *ClusterQueue) IsBorrowing() bool {
```

Comment:
we should probably move all the ClusterQueue code to a different file, but for the future :sweat_smile:

Reply:
Agree.

Reply:
Through more work, there is a struct, `Cohort`, that both `Cache` and `ClusterQueue` have.
I think we should remove `cohort` from `Cache`.

@alculquicondor wdyt?

Reply:
I think multiple structs that have the same info will cause potential issues, such as forgetting to update either cohort.

Reply:
yeah, we can move Cohort too.

> I think multiple structs that have the same info will cause potential issues, such as forgetting to update either cohort.

Not sure what same info you are referring to.

Reply:
@alculquicondor I meant we remove `cohorts` from  `Cache` since we can get cohorts from `Cache.clusterQueues.Cohort`.

https://github.com/kubernetes-sigs/kueue/blob/c664e87ae90b4fd2a0a933224435ef6c04a115b2/pkg/cache/cache.go#L74C1-L75C1

Actually, we do the same operation to both `Cache.Cohort` and `Cache.clusterQueues.cohorts` like this:

https://github.com/kubernetes-sigs/kueue/blob/c664e87ae90b4fd2a0a933224435ef6c04a115b2/pkg/cache/cache.go#L962-L968

https://github.com/kubernetes-sigs/kueue/blob/c664e87ae90b4fd2a0a933224435ef6c04a115b2/pkg/cache/cache.go#L975-L980

Reply:
if it's possible to remove without losing information or increasing lookup time, go for it.

Reply:
The `Cache.cohorts` is used only by `Snapshot`.

https://github.com/kubernetes-sigs/kueue/blob/20b82b2acaabc9ac91156bde6ba7095e30f75a52/pkg/cache/snapshot.go#L75-L85

So I think we can remove `Cache.cohorts` by refactoring this part without increasing lookup time.

Reply:
sg, give it a try